pkg/rpc/client/rpc.go

```diff
@@ -453,8 +450,7 @@ func (c *Client) SignAndPushInvocationTx(script []byte, acc *wallet.Account, sys
 		return txHash, errors.Wrap(err, "failed to sign tx")
 	}
 	txHash = tx.Hash()
-	err = c.SendRawTransaction(tx)
-
+	_, err = c.SendRawTransaction(tx)
```

Comment:
So what if it's different from `txHash`?

Reply:
Well, there was a check, but https://github.com/nspcc-dev/neo-go/pull/1216#discussion_r457908438. So, should we also return an error in this case?

Reply:
It shouldn't ever happen, but it can. And if it can, it must be handled in some way. At the moment you're just ignoring this potential mismatch and returning `txHash`, but what's the value of this `txHash` for the user if it doesn't match server's hash? And it can happen easily with misconfigured client, when you're using some privnet client for mainnet network, for example.

We can of course just not compute this hash client-side, relying on server's answer. Theoretically it would allow us to be network-agnostic on the client, but I don't think we could do it for all the calls and we require proper network setup for the client anyway for everything to function correctly.

So I think we should add this safety check, just in case.

Reply:
Added check, now `SignAndPushInvocationTx` returns an error containing both hashes in case when hashes mismatch.